src/plugins/timeConductor/ConductorClock.vue

```diff
@@ -0,0 +1,128 @@
+/***************************************************************************** * Open MCT Web,
+Copyright (c) 2014-2023, United States Government * as represented by the Administrator of the
+National Aeronautics and Space * Administration. All rights reserved. * * Open MCT Web is licensed
+under the Apache License, Version 2.0 (the * "License"); you may not use this file except in
+compliance with the License. * You may obtain a copy of the License at *
+http://www.apache.org/licenses/LICENSE-2.0. * * Unless required by applicable law or agreed to in
+writing, software * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT *
+WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the * License for the specific
+language governing permissions and limitations * under the License. * * Open MCT Web includes source
+code licensed under additional open source * licenses. See the Open Source Licenses file
+(LICENSES.md) included with * this source code distribution or the Licensing information page
+available * at runtime from the About dialog for additional information.
+*****************************************************************************/
+<template>
+  <div v-if="readOnly === false" ref="clockButton" class="c-tc-input-popup__options">
+    <div class="c-menu-button c-ctrl-wrapper c-ctrl-wrapper--menus-left">
+      <button
+        class="c-button--menu c-button--compact js-clock-button"
+        :class="[buttonCssClass, selectedClock.cssClass]"
+        @click.prevent.stop="showClocksMenu"
+      >
+        <span class="c-button__label">{{ selectedClock.name }}</span>
+      </button>
+    </div>
+  </div>
+  <div v-else class="c-compact-tc__clock">
+    <div class="c-compact-tc__clock__value">{{ selectedClock.name }}</div>
+  </div>
+</template>
+
+<script>
+import modeMixin from './mode-mixin';
+import { TIME_CONTEXT_EVENTS } from '../../api/time/constants';
+
+export default {
+  mixins: [modeMixin],
```

Comment:
Hmm. It doesn't look like there is much overlap between the clock menu and the mode menu. Could we separate the mode-mixin into two mixins? One for clocks and one for modes?

Reply:
Yep! Definitely. They diverged quite a bit... not any overlap I think :)